### PR TITLE
feat: support bech32 addresses

### PR DESCRIPTION
**Pull request by joroshiba**

### CODE REVIEW

File: params/config.go [hunk 1]
```diff
@@ -1063,9 +1066,18 @@ type AstriaErc20AssetConfig struct {
 	ContractPrecision uint16         `json:"contractPrecision"`
 }
 
-func (abc *AstriaBridgeAddressConfig) Validate() error {
-	if len(abc.BridgeAddress) != 20 {
-		return fmt.Errorf("bridge address must be 20 bytes")
+func (abc *AstriaBridgeAddressConfig) Validate(hrpPrefix string) error {
+	prefix, byteAddress, err := bech32.Decode(abc.BridgeAddress)
+	byteAddress, _ = bech32.ConvertBits(byteAddress, 5, 8, false)
```
Comment (noot) [inline params/config.go:1071]: don't ignore the error here?

File: params/config.go [hunk 2]
```diff
@@ -1063,9 +1066,18 @@ type AstriaErc20AssetConfig struct {
 	ContractPrecision uint16         `json:"contractPrecision"`
 }
 
-func (abc *AstriaBridgeAddressConfig) Validate() error {
-	if len(abc.BridgeAddress) != 20 {
-		return fmt.Errorf("bridge address must be 20 bytes")
+func (abc *AstriaBridgeAddressConfig) Validate(hrpPrefix string) error {
+	prefix, byteAddress, err := bech32.Decode(abc.BridgeAddress)
```
Comment (noot) [inline params/config.go:1070]: check error here

File: grpc/execution/test_utils.go [hunk 3]
```diff
@@ -48,14 +50,21 @@ func generateMergeChain(n int, merged bool) (*core.Genesis, []*types.Block, *ecd
 		panic(err)
 	}
 	bridgeAddress := crypto.PubkeyToAddress(bridgeAddressKey.PublicKey)
+	bridgeAddressBytes, err := bech32.ConvertBits(bridgeAddress.Bytes(), 8, 5, false)
```
Comment (bharath-123) [inline grpc/execution/test_utils.go:53]: better to catch errors in tests too.

File: grpc/execution/validation_test.go [hunk 4]
```diff
@@ -32,6 +34,25 @@ func testDepositTx() *types.Transaction {
 	})
 }
 
+func generateBech32MAddress() string {
+	addressKey, err := crypto.GenerateKey()
+	if err != nil {
+		panic(err)
```
Comment (bharath-123) [inline grpc/execution/validation_test.go:40]: instead of panicking, we can return an error and throw a `t.Fatal` where we use this method.

**Review by bharath-123 — APPROVED** [review]

File: grpc/execution/validation_test.go (hunk 4 above)
Comment (joroshiba) [reply grpc/execution/validation_test.go:40]: I don't know why this will ever happen, copied how we behave elsewhere in the tests when generating a key

**Pull request merged**